test/active_record_doctor/detectors/missing_unique_indexes_test.rb

```diff
@@ -179,12 +179,58 @@ def test_column_order_is_ignored
     refute_problems
   end
 
-  def test_conditions_is_skipped
-    assert_skipped(conditions: -> { where.not(email: nil) })
+  def test_case_insensitive_index_exists
+    skip("Expression indexes are not supported") if expression_indexes_unsupported?
+
+    create_table(:users) do |t|
+      t.string :email
+
+      t.index :email, unique: true
+    end.create_model do
+      validates :email, uniqueness: { case_sensitive: false }
+    end
+
+    assert_problems(<<~OUTPUT)
+      add a unique index on users(lower(email)) - validating uniqueness in the model without an index can lead to duplicates
```

Comment:
I suggest a different error message when the complaint is about a missing **expression** index:

```
add a unique expression index on users(lower(email)) - validating case-insensitive uniqueness in the model without an expression index can lead to duplicates (a regular unique index is not enough)
```

Reply:
Also, we need another test case that creates a **non-unique expression index** on the column.

The index must satisfy two conditions to satisfy the detector:

1. Unique
2. Expression

This gives four combinations in total:

1. No index at all (already covered by a previous test case).
2. Unique, no expression (the test case you added).
3. Non-unique, expression (the test case that's missing).
4. Unique, expression (added below).